### ISSUE TITLE
New Design and Adding Several Stuffs

### ISSUE THREAD

**Issue by wisn** [haskell-id/website#5]
I think we need to update the web design and adding several features. We could discuss how it looks. As for new feature I mentioned, how about blog? Nah, contributors could send a PR for their blog post title, description or summary, and its URL. Yep, the actual content is on their blog. Also, any suggestions for newcomer guide related content?

**Comment by kusut** [comment]
Ok. Here's my take

I'm all for new looks. I'm bad at css, so I tried to be minimal. However, I have a strong opinion on fast and light website. People with poor connection should have no problem accessing the site. Use [this](https://www.webpagetest.org), chrome lighthouse, or any tool to achieve this result.

I dont mind getting new features. This site runs on small number of volunteers with limited time, so we're trying to be sensible and minimize maintenance work.

TLDR for future reference.
- It should be fast (since we're only serving static content at the moment). 
- Be wary of maintenance work. Dont bite off more than we could chew. Adding more static content (blog links) should be fine. However, turning this site into blogging platform will bring a lot of maintenance work. It is your call (since I made you an owner).

Btw, this site is running on my firebase account for now. I will try to redeploy any changes daily.

cc @matematikaadit @haskell-id/admins 

Happy hacking!

**Comment by matematikaadit** [comment]
I'm mostly agree with uncle @kusut  :smile: 

Some more addition, maybe from accessibility point:

- Mobile friendly.
- Good contrast and font-size. 

My main complain for most modern website is usually, they look bad on mobile, or they have very low contrast and small font-size. Not all people have good sight, especially when they got older :laughing:

**Comment by edom** [comment]
I totally agree with @kusut . Minimize maintenance. Be lazy like Haskell. Avoid doing unnecessary things. :smile: 

In my opinion, the current design is fine (I like uncluttered fast-loading minimal websites like this). I'm OK if anyone wants to change it, as long as it doesn't reduce the website's usability, but I perceive no clear benefits from changing the style. I think any styling beyond this is bikeshedding, so I'm going to neither encourage nor impede anyone who wants to do that. If you want to do it, feel free. If you don't want to do it, great.

https://phinze.github.io/2014/05/24/useful-tech-terms-part-1.html

It's OK for us to link to others' blogs. However, turning the site into a blog might be too much for now.

@kusut I think the website is running on GitHub pages? https://github.com/haskell-id/haskell-id.github.io

> Also, any suggestions for newcomer guide related content?

How to install Haskell? Stephen Diehl's "What I wish I knew about Haskell"? What content did you wish you had when you were starting out? Task-oriented documentation?

I think people will appreciate content more than styling (unless the style is atrocious). :smile: 

@matematikaadit I shrank my website font size from 12pt to 10pt because I want to fit more text on screen. I assume that the user knows about Ctrl-+ but I may be wrong. :smile:

**Comment by matematikaadit** [comment]
> @matematikaadit I shrank my website font size from 12pt to 10pt because I want to fit more text on screen. I assume that the user knows about Ctrl-+ but I may be wrong. :smile:

@edom depending on the audience itself, and how the content being consumed. I think on mobile there's no easy way to resize the font (pinch zoom didn't count since it also resize the page's width). [This is my favorite article](http://mrmrs.cc/writing/2016/03/23/the-veil-of-ignorance/) regarding this topic (might also related to kusut's request for it to be fast and light). Anyway, good stuff there :+1: , offtopic, but are you using Org-Mode on Emacs?

**Comment by edom** [comment]
@matematikaadit On mobile I feel 10pt is good and 12pt is too big (pt not px). My vision is not 20/20 either (I can only read 12pt text from about 40cm). Your favorite article makes me doubt my choices once again... (offtopic:) Thanks. No, I'm not using org-mode. I'm using Visual Studio Code, Markdown, and Jekyll :speak_no_evil:.

**Comment by kusut** [comment]
@edom No. I switched to firebase because I wanted to try things out. 

Current build (this needs to be on readme or something once we clean this up): each commit (unless there's a skip message) will:
- Trigger travis ci build. Stack based. This is pushed to haskell-id.github.io
- Trigger circleci build. cabal-new based. (I plan to do multi ghc build here if I ever have the time. Seems easier than travis last time I check).
- haskell.web.id deployment is not automated. I deploy with my laptop to my firebase account.

So haskell-id.github.io right now is NOT haskell.web.id. This breaks my own rule about maintenance. I'll probably switch back to github pages, or netlify, or other service and make it automated. Any thoughts? Github pages looks easier ya? [Although I want it to be a single repo if we go that way](http://blog.jenkster.com/2016/02/git-for-static-sites.html)

cc @matematikaadit @wisn

**Comment by kusut** [comment]
@matematikaadit does the nix on readme still work? Got syntax error when trying it out with latest nix few months ago

**Comment by matematikaadit** [comment]
@kusut 
> So haskell-id.github.io right now is NOT haskell.web.id. This breaks my own rule about maintenance. I'll probably switch back to github pages, or netlify, or other service and make it automated. Any thoughts? Github pages looks easier ya?

I have no particular preference. Anything is okay for me as long as it's working.

> Although I want it to be a single repo if we go that way

+1 for single repo.

> @matematikaadit does the nix on readme still work? Got syntax error when trying it out with latest nix few months ago

Unless the package name in the nixpkgs changed, or there's significant change in the syntax, it should works as expected. I'll investigate them in the evening.

@wisn 
I hope you're not discouraged. It's always a good things to have more people as contributor.

**Comment by wisn** [comment]
@edom 

> How to install Haskell? Stephen Diehl's "What I wish I knew about Haskell"? What content did you wish you had when you were starting out? Task-oriented documentation?

I think Haskell installation had in there. Something like "What I wish I knew about Haskell" seems cool.

> I think people will appreciate content more than styling (unless the style is atrocious). :smile:

I will try to design first then let see. Well, if the new design is "bad" we could just add new "feature" instead.

@kusut 

> So haskell-id.github.io right now is NOT haskell.web.id. This breaks my own rule about maintenance. I'll probably switch back to github pages, or netlify, or other service and make it automated. Any thoughts? Github pages looks easier ya?

I'm with you. I personally prefer GitHub Pages and we should think about its deployment automation. Hence, Netlify also good for me. Well, as long as there is automation for the deployment process.

> Although I want it to be a single repo if we go that way

+1

@matematikaadit 

>  I hope you're not discouraged. It's always a good things to have more people as contributor.

No worries.

**Comment by kusut** [comment]
Oh. FYI, switching to firebase helps with webpagetest score since we can set Cache-Control for static content. Also github didn't have https back then.

I'll keep trying to find easier deployment solution. We'll be back to github when push come to shove

**Comment by edom** [comment]
@wisn I think it would also be cool if we have "Haskell for Java programmers", "Haskell for Python programmers", "Haskell for Ruby programmers", and so on, so that people can learn Haskell more quickly with their past knowledge (a.k.a. "transfer learning").

@kusut I use two repos (one source, one GitHub Pages HTML), and I kinda nuke the `*.github.io` repository every time I push, because I didn't know about worktrees (and my Git version is rather ancient) :joy::

https://github.com/edom/work/blob/master/sh/pub

My OS seems to be too old (Ubuntu 14.04 comes with Git 1.9.1; worktrees are introduced in Git 2.5), but it shouldn't be too hard to install a newer version of Git to `~/.local/bin`. But feel free if you want to use one repo.

I think after GitHub can do HTTPS for custom domains, GitHub is the minimum-maintenance way to deploy static websites. But, as you say, GitHub doesn't have Cache-Control. But the benefit of Cache-Control may be small because haskell.web.id is 18.7 KB without cache :+1:. But, for SEO, page speed does matter. (We're going to choose minimum-maintenance over SEO, aren't we? :smile: )

I tried to set up multi-GHC build with cabal new-install on travisci, but I gave up. The travisci script uses cabal install, and it clashes with cabal new-install. Sometimes cabal even runs out of RAM on travisci. I'm going to wait until cabal new-install becomes mainstream.

https://github.com/edom/work/blob/travis/.travis.yml
https://travis-ci.com/edom/work/jobs/134206459
